src/main/java/org/ow2/proactive/catalog/repository/BucketGrantRepository.java

```diff
@@ -0,0 +1,71 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.catalog.repository;
+
+import java.util.List;
+
+import javax.persistence.LockModeType;
+import javax.persistence.QueryHint;
+
+import org.ow2.proactive.catalog.repository.entity.BucketGrantEntity;
+import org.springframework.data.jpa.repository.*;
+import org.springframework.data.querydsl.QueryDslPredicateExecutor;
+
+
+public interface BucketGrantRepository extends JpaRepository<BucketGrantEntity, Long>,
+        JpaSpecificationExecutor<BucketGrantEntity>, QueryDslPredicateExecutor<BucketGrantEntity> {
+
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @QueryHints({ @QueryHint(name = "javax.persistence.lock.timeout", value = "5000") })
+    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.bucketEntity.id = ?1")
+    List<BucketGrantEntity> findAllGrantsByBucket(long bucketId);
+
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @QueryHints({ @QueryHint(name = "javax.persistence.lock.timeout", value = "5000") })
+    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.bucketEntity.id = ?1 AND bge.profiteer = ?2 AND bge.grantee='user'")
+    BucketGrantEntity findBucketGrantByUsername(long bucketId, String username);
+
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @QueryHints({ @QueryHint(name = "javax.persistence.lock.timeout", value = "5000") })
+    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.bucketEntity.id = ?1 AND bge.profiteer = ?2 AND bge.grantee='group'")
+    BucketGrantEntity findBucketGrantByUserGroup(long bucketId, String userGroup);
+
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @QueryHints({ @QueryHint(name = "javax.persistence.lock.timeout", value = "5000") })
+    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.creator = ?1")
+    List<BucketGrantEntity> findAllGrantsCreatedByUsername(String admin);
+
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @QueryHints({ @QueryHint(name = "javax.persistence.lock.timeout", value = "5000") })
+    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.profiteer = ?1 And bge.grantee='user'")
+    List<BucketGrantEntity> findAllGrantsAssignedToAUsername(String username);
+
+    @Lock(LockModeType.PESSIMISTIC_WRITE)
+    @QueryHints({ @QueryHint(name = "javax.persistence.lock.timeout", value = "5000") })
+    @Query(value = "SELECT bge FROM BucketGrantEntity bge WHERE bge.profiteer = ?1 And bge.grantee='group'")
+    List<BucketGrantEntity> findAllGrantsAssignedToAUserGroup(String userGroup);
+
+}
```

Comment:
I think that you can shorten and improve the readability of all the jpaRepo classes using JPA query method as explained here: https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query-methods.query-creation